File: 0653-two-sum-iv-input-is-a-bst/0653-two-sum-iv-input-is-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool findTarget(TreeNode* root, int k) {
        vector<int> res;
        inorder(root, res);
        int i=0;
        int j=res.size()-1;
        while(i<j){
            if(res[i]+res[j]==k){
                return true;
            }else if(res[i]+res[j]<k){
                i++;
            }else{
                j--;
            }
        }
        return false;
    }
    void inorder(TreeNode* root, vector<int>& res){
        if(!root){
            return;
        }
        inorder(root->left, res);
        res.push_back(root->val);
        inorder(root->right, res);
    }
};

File: 0653-two-sum-iv-input-is-a-bst/0653-two-sum-iv-input-is-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findTarget(self, root: Optional[TreeNode], k: int) -> bool:
        res=[]
        self.inorder(root, res)
        i=0
        j=len(res)-1
        while i<j:
            if res[i]+res[j]==k:
                return True
            elif res[i]+res[j]<k:
                i+=1
            else:
                j-=1
    
        return False;

    
    def inorder(self, root, res):
        if root==None:
            return

        self.inorder(root.left, res)
        res.append(root.val)
        self.inorder(root.right, res)